trestle/core/control_interface.py

```diff
@@ -0,0 +1,616 @@
+        # assumes at most one instance
+        names = [prop.name for prop in as_list(part_control.props)]
+        if new_prop.name in names:
+            index = names.index(new_prop.name)
+            del part_control.props[index]
+        part_control.props = as_list(part_control.props)
+        part_control.props.append(new_prop)
+
+    @staticmethod
+    def get_sort_id(control: cat.Control, allow_none=False) -> Optional[str]:
+        """Get the sort-id for the control."""
+        for prop in as_list(control.props):
+            if prop.name == const.SORT_ID:
+                return prop.value.strip()
+        return None if allow_none else control.id
+
+    @staticmethod
+    def get_label(part_control: TypeWithProps) -> str:
+        """Get the label from the props of a part or control."""
+        return ControlInterface.get_prop(part_control, 'label')
+
+    @staticmethod
+    def get_part(part: common.Part, item_type: str, skip_id: Optional[str]) -> List[Union[str, List[str]]]:
+        """
+        Find parts with the specified item type, within the given part.
+
+        For a part in a control find the parts in it that match the item_type
+        Return list of string formatted labels and associated descriptive prose
+        """
+        items = []
+        if part.name in ['statement', item_type]:
+            # the options here are to force the label to be the part.id or the part.label
+            # the label may be of the form (a) while the part.id is ac-1_smt.a.1.a
+            # here we choose the latter and extract the final element
+            label = ControlInterface.get_label(part)
+            label = part.id.split('.')[-1] if not label else label
+            wrapped_label = ControlInterface._wrap_label(label)
+            pad = '' if wrapped_label == '' or not part.prose else ' '
+            prose = '' if part.prose is None else part.prose
+            # top level prose has already been written out, if present
+            # use presence of . in id to tell if this is top level prose
+            if part.id != skip_id:
+                items.append(f'{wrapped_label}{pad}{prose}')
+            if part.parts:
+                sub_list = []
+                for prt in part.parts:
+                    sub_list.extend(ControlInterface.get_part(prt, item_type, skip_id))
+                sub_list.append('')
+                items.append(sub_list)
+        return items
+
+    @staticmethod
+    def get_adds_for_control(profile: prof.Profile, control_id: str) -> List[prof.Add]:
+        """Get the adds for a given control id from a profile."""
+        adds: List[prof.Add] = []
+        if profile.modify:
+            for alter in as_list(profile.modify.alters):
+                if alter.control_id == control_id:
+                    adds.extend(as_list(alter.adds))
+        return adds
+
+    @staticmethod
+    def get_adds(control_id: str, profile: prof.Profile) -> List[Tuple[str, str]]:
```

Comment:
Shouldn't be it named get prose for adds or something?

Reply:
changed to get_all_add_prose